autoblocks/_impl/tracer.py

```diff
@@ -101,38 +109,120 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    def _evaluation_to_json(evaluation: Evaluation, evaluator_external_id: str) -> Dict[str, Any]:
+        return dict(
+            id=str(uuid.uuid4()),
+            score=evaluation.score,
+            threshold=dataclasses.asdict(evaluation.threshold) if evaluation.threshold else None,
+            metadata=evaluation.metadata,
+            evaluatorExternalId=evaluator_external_id,
+        )
+
+    async def _evaluate_event(self, event: TracerEvent, evaluator: BaseEventEvaluator) -> Optional[Evaluation]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                log.error(f"Unable to execute evaluator with id: {evaluator.id}. Error: %s", err, exc_info=True)
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                log.error(f"Unable to execute evaluator with id: {evaluator.id}. Error: %s", err, exc_info=True)
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: List[BaseEventEvaluator], event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[Dict[str, Any]]:
+        if len(evaluators) == 0:
+            return []
+        try:
```

Comment:
since you're already try-excepting up top you could probably drop this one and rename `_run_evaluators` to `_run_evaluators_unsafe` to follow the same pattern as send event. this basically lets you write a "clean" function w/o any error handling and then push the try-except to the outer call